src/main/ruby/core/process.rb

```diff
@@ -551,7 +551,8 @@ def self.wait2(input_pid=-1, flags=nil)
         exitcode, termsig, stopsig = ptr.read_array_of_int(3).map { |e| e == -1000 ? nil : e }
 
         status = Process::Status.new(pid, exitcode, termsig, stopsig)
-        $? = status
+        locals = Truffle.invoke_primitive :thread_get_locals, Thread.current
+        Truffle.invoke_primitive :object_ivar_set, locals, :$?, status
```

Comment:
This seems like a common pattern. Should it be a method?

Reply:
Yeah, I think there probably should be a method on 'Truffle::ThreadOperarions' or similar.